exasol_sphinx_github_pages_generator/deploy_github_pages.py

```diff
@@ -1,150 +1,15 @@
-    # Builds the Sphinx documentation. Generates html files inside "build_dir" using "source_dir"
-    # -W: Turns warnings into errors
-    run(["sphinx-build", "-b", "html", "-d", intermediate_dir, "-W", source_dir, build_dir], check=True)
-    print("Generated HTML Output")
-    print(f"Using html_output_dir={build_dir}")
-    run(["ls", "-la", build_dir], check=True)
-
-    output_dir = Path(worktree + "/" + source_branch)
-    print(f"Using output_dir={output_dir}")
-    if output_dir.exists() and output_dir.is_dir():
-        print(f"Removing existing output directory {output_dir}")
-        shutil.rmtree(output_dir)
-    print(f"(Re)Creating output directory {output_dir}")
-    output_dir.mkdir(parents=True)
-    print(f"Copying HTML output {build_dir} to the output directory {output_dir}")
-    for obj in os.listdir(build_dir):
-        shutil.move(build_dir + "/" + str(obj), output_dir)
-    print(f"Content of output directory {output_dir}")
-    open(f"{worktree}/.nojekyll", "w").close()
-    run(["ls", "-la", output_dir], check=True)
-    return output_dir
-
-
-def git_commit_and_push(worktree, push_origin, push_enabled, source_branch, output_dir, current_commit_id,
-                            target_branch):
-    currentworkdir = os.getcwd()
-    print(currentworkdir)
-    os.chdir(worktree)
-    print(f"Current directory before commit and push {os.getcwd()}")
-    print("Git commit")
-    with open(f"{output_dir}/.source", "w+") as file:
-        file.write(f"BRANCH={source_branch} \n")
-        file.write(f"COMMIT_ID={current_commit_id}")
-    run(["git", "add", "-v", "."], check=True)
-    changes_exists = run(["git", "diff-index", "--quiet", "HEAD", "--"], capture_output=True, text=True)
-    if 1 == changes_exists.returncode:
-        print(f"committing changes because changes_exist is {changes_exists.returncode}")
-        run(["git", "commit", "--no-verify", "-m",
-             f"Update documentation from source branch {source_branch} with commit id {current_commit_id}"], check=True)
-        if push_origin != "" and push_enabled == "push":
-            print(f"Git push {push_origin} {target_branch}")
-            run(["git", "push", push_origin, target_branch], check=True)
-    elif 0 == changes_exists.returncode:
-        print("No changes to commit.")
-    else:
-        print('A Error occurred in run(["git", "diff-index", "--quiet", "HEAD", "--"], capture_output=True, text=True)')
-        print(f" changes_exists.returncode: {changes_exists.returncode}")
-        print(f" changes_exists.stderr: {changes_exists.stderr}")
-        print(f" changes_exists.stdout: {changes_exists.stdout}")
-    os.chdir(currentworkdir)
-
-
-def clean_worktree(worktree):
-    wt = Path(worktree)
-    if wt.exists():
-        print(f"Cleanup git worktree {worktree}")
-        run(["git", "worktree", "remove", "--force", worktree], check=True)
-
 
 def deploy_github_pages(argv):
     args = Parser(argv).args
-
+    original_workdir = os.getcwd()
```

Comment:
Yes, but i don't know if it would really add to the readability. Especially because we still need the worktree cleanup function. and this would either hide it, or duplicate the try/finally block.

Reply:
Besides the readability I think it simplifies the code  ("it is separation of concerns") because you get rid of dealing with going in and out of the cwd by just doing/using a ctx-mgr (details for that are hidden in the ctx-mgr). But do what are you feeling more comfortable with and what does make more sense to you.